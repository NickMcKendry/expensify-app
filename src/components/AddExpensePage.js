import React from 'react';
import ReactDOM from 'react-dom';

const AddExpensePage = () => (
  <div>
    <h3>This is add</h3>
  </div>
);

export default AddExpensePage;
