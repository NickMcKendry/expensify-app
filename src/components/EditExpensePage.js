import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';

const EditExpensePage = (props) => {
  return (
    <div>
      <ExpenseForm
        expense={props.expense}
        onSubmit={(expense) => {
          props.dispatch(editExpense(props.match.params.id, expense));
          props.history.push('/');
        }}
      />
      <button onClick={() => {
        props.dispatch(removeExpense({ id: props.match.params.id }));
        props.history.push('/');
        console.log(`removed ${props.match.params.id}`);
      }}>Remove</button>
    </div>
  )
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
}

export default connect(mapStateToProps)(EditExpensePage);
