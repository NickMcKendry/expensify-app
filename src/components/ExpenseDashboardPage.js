import React from 'react';
import ReactDOM from 'react-dom';

const ExpenseDashboardPage = () => (
  <div>
    <h3>This is dashboard</h3>
  </div>
);

export default ExpenseDashboardPage;
