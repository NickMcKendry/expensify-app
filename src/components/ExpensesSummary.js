import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import getExpensesTotal from '../selectors/expenses-total';
import getVisibleExpenses from '../selectors/expenses';


export const ExpensesSummary = ({ expensesTotal, expenseCount }) => (
  <div>
    <h3>Showing {expenseCount === 1 ? `${expenseCount} expense` :`${expenseCount} expenses` } with a total of {numeral(expensesTotal / 100).format('$0,0.00')}</h3>
  </div>
)






const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: getExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
