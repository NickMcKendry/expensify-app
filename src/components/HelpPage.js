import React from 'react';
import ReactDOM from 'react-dom';

const HelpPage = () => (
  <div>
    <h3>This is Help</h3>
  </div>
);

export default HelpPage;
