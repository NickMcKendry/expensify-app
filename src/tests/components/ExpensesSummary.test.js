import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render single expense summary', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={150} />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render multiple expenses summary', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={250} />);
  expect(wrapper).toMatchSnapshot();
});
