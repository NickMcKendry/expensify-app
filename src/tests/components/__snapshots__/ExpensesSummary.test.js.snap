// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render multiple expenses summary 1`] = `
<div>
  <h3>
    Showing 
    2 expenses
     with a total of 
    $2.50
  </h3>
</div>
`;

exports[`Should render single expense summary 1`] = `
<div>
  <h3>
    Showing 
    1 expense
     with a total of 
    $1.50
  </h3>
</div>
`;
